File: src/Implicit/ImplicitFloor.cs
using System;

namespace TinkerWorX.AccidentalNoiseLibrary
{
    public sealed class ImplicitFloor : ImplicitModuleBase
    {
        public ImplicitFloor()
        {
            this.Source = new ImplicitConstant(0.00);
        }

        public ImplicitFloor(ImplicitModuleBase source)
        {
            this.Source = source;
        }

        public ImplicitModuleBase Source { get; set; }

        public override double Get(double x, double y)
        {
            return Math.Floor(this.Source.Get(x, y));
        }

        public override double Get(double x, double y, double z)
        {
            return Math.Floor(this.Source.Get(x, y, z));
        }

        public override double Get(double x, double y, double z, double w)
        {
            return Math.Floor(this.Source.Get(x, y, z, w));
        }

        public override double Get(double x, double y, double z, double w, double u, double v)
        {
            return Math.Floor(this.Source.Get(x, y, z, w, u, v));
        }
    }
}

File: src/Implicit/ImplicitFloor.java
package Implicit;

import util.Maths;

public final class ImplicitFloor extends ImplicitModuleBase {
    
    private ImplicitModuleBase Source;
    
    public ImplicitFloor() {
        this.Source = new ImplicitConstant(0.00);
    }
    
    public ImplicitFloor(final ImplicitModuleBase source) {
        this.Source = source;
    }
    
    public ImplicitModuleBase getSource() {
        return this.Source;
    }
    
    public void setSource(final ImplicitModuleBase source) {
        this.Source = source;
    }
    
    @Override
    public double get(final double x, final double y) {
        return Maths.floor(this.Source.get(x, y));
    }
    
    @Override
    public double get(final double x, final double y, final double z) {
        return Maths.floor(this.Source.get(x, y, z));
    }
    
    @Override
    public double get(final double x, final double y, final double z, final double w) {
        return Maths.floor(this.Source.get(x, y, z, w));
    }
    
    @Override
    public double get(final double x, final double y, final double z, final double w, final double u, final double v) {
        return Maths.floor(this.Source.get(x, y, z, w, u, v));
    }
}
